server/app/views/admin/programs/ProgramApplicationListView.java

```diff
@@ -95,31 +114,135 @@ public Content render(
+                            .setLabelText("Download all data")
+                            .setValue("1")
+                            .setChecked(false)
+                            .getContainer(),
+                        input()
+                            .withName(FROM_DATE_PARAM)
+                            .withValue(filterParams.fromDate().orElse(""))
+                            .withType("hidden"),
+                        input()
+                            .withName(UNTIL_DATE_PARAM)
+                            .withValue(filterParams.untilDate().orElse(""))
+                            .withType("hidden"),
+                        input()
+                            .withName(SEARCH_PARAM)
+                            .withValue(filterParams.search().orElse(""))
+                            .withType("hidden"),
+                        div()
+                            .withClasses(Styles.FLEX, Styles.MT_6, Styles.SPACE_X_2)
+                            .with(
+                                TagCreator.button("Download CSV")
+                                    .withClasses(
+                                        ReferenceClasses.DOWNLOAD_ALL_BUTTON,
+                                        AdminStyles.PRIMARY_BUTTON_STYLES)
+                                    .attr(
+                                        Attr.FORMACTION,
+                                        controllers.admin.routes.AdminApplicationController
+                                            .downloadAll(
+                                                program.id(),
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                Optional.empty())
+                                            .url())
+                                    .withType("submit"),
+                                TagCreator.button("Download JSON")
+                                    .withClasses(
+                                        ReferenceClasses.DOWNLOAD_ALL_BUTTON,
+                                        AdminStyles.PRIMARY_BUTTON_STYLES)
+                                    .attr(
+                                        Attr.FORMACTION,
+                                        controllers.admin.routes.AdminApplicationController
+                                            .downloadAllJson(
+                                                program.id(),
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                Optional.empty())
+                                            .url())
+                                    .withType("submit"))));
+    return Modal.builder(modalId, modalContent)
```

Comment:
Should we disable downloads when the current filters haven't been applied?   (or give some indication that the changes to filters won't be reflected in the download)